lib/ansible/errors/__init__.py

```diff
@@ -173,6 +173,19 @@ class AnsibleConnectionFailure(AnsibleRuntimeError):
     ''' the transport / connection_plugin had a fatal error '''
     pass
 
+class AnsibleSshConnectionFailure(AnsibleRuntimeError):
```

Comment:
it should be subclass of AnsibleConnectionFailure at least to be able to handle these genericly